src/docs/clustering/cluster-client.md

```diff
@@ -1,42 +1,246 @@
+}, host1);
+
+RunOn(() =>
+{
+    var serviceB = Sys.ActorOf(Props.Create<Service>(), "serviceB");
+    ClusterClientReceptionist.Get(Sys).RegisterService(serviceB);
+}, host2, host3);
+```
+
+On the client you create the `ClusterClient` actor and use it as a gateway for sending messages to the actors identified by their path (without address information) somewhere in the cluster.
 
 ```csharp
-// inside the Cluster
-var receptionist = ClusterClientReceptionist.Get(system);
-receptionist.RegisterService(actorRef);
+RunOn(() =>
+{
+    var c = Sys.ActorOf(Client.ClusterClient.Props(
+        ClusterClientSettings.Create(Sys).WithInitialContacts(initialContacts)), "client");
+    c.Tell(new Client.ClusterClient.Send("/user/serviceA", "hello", localAffinity: true));
+    c.Tell(new Client.ClusterClient.SendToAll("/user/serviceB", "hi"));
+}, client);
+```
+The `initialContacts` parameter is a `IEnumerable<ActorPath>`, which can be created like this:
+
+```csharp
+var initialContacts = new List<ActorPath>()
+{
+    ActorPath.Parse("akka.tcp://OtherSys@host1:2552/system/receptionist"),
+    ActorPath.Parse("akka.tcp://OtherSys@host2:2552/system/receptionist")
+};
+
+var settings = ClusterClientSettings.Create(Sys).WithInitialContacts(initialContacts);
+```
+
+You will probably define the address information of the initial contact points in configuration or system property. See also Configuration.
+
+## ClusterClientReceptionist Extension
+In the example above the receptionist is started and accessed with the `Akka.Cluster.Tools.Client.ClusterClientReceptionist` extension. That is convenient and perfectly fine in most cases, but it can be good to know that it is possible to start the `akka.cluster.client.ClusterReceptionist` actor as an ordinary actor and you can have several different receptionists at the same time, serving different types of clients.
+
+Note that the `ClusterClientReceptionist` uses the `DistributedPubSub` extension, which is described in Distributed Publish Subscribe in Cluster.
+
+It is recommended to load the extension when the actor system is started by defining it in the akka.extensions configuration property:
 
-// outside the cluster
-system.Settings.InjectTopLevelFallback(ClusterClientReceptionist.DefaultConfig());
-var settings = ClusterClientSettings.Create(system);
-var client = system.ActorOf(ClusterClient.Props(settings));
+```hocon
+akka.extensions = ["Akka.Cluster.Tools.Client.ClusterClientReceptionist"]
+```
+
+## Events
+As mentioned earlier, both the `ClusterClient` and `ClusterClientReceptionist` emit events that can be subscribed to. The following code snippet declares an actor that will receive notifications on contact points (addresses to the available receptionists), as they become available. The code illustrates subscribing to the events and receiving the `ClusterClient` initial state.
+
+```scala
+class ClientListener(targetClient: ActorRef) extends Actor {
```

Comment:
Could port this examples at the moment. I should update ClusterClient code first